ccplugin/RpcHelper.cs

```diff
@@ -238,7 +235,7 @@ public static byte[] ReadProtobuf(HttpClient httpClient, string url, out string
             if (dealStateObj.ContainsKey(ERROR))
             {
                 msg = (string)dealStateObj[ERROR][MESSAGE];
-                return null;
+                return new byte[0];
```

Comment:
For this warning (Return an empty collection instead of null.) Sonar says

> Returning null instead of an actual array or collection forces callers of the method to explicitly test for nullity, making them more complex and less readable.

I don't see a way to make a logically equivalent change here.  If returning `null` is the desired behavior then we should make Sonar ignore this.

Reply:
@vladimir-gluwa, @will-gluwa what's the preference here? Leave the code as-is or satisfy Sonar and introduce a backwards incompatible change ?

Reply:
This is up to @vladimir-gluwa

Reply:
ther are three valid states: a collection containing data, an empty collection and input error, null is designaterd to convey the third state, otherwise it will have to be conveyed by other means and complexity and readability won't be improved. please revert the change